drivers/include/periph/uart.h

```diff
@@ -97,6 +97,7 @@ typedef void(*uart_rx_cb_t)(void *arg, uint8_t data);
 typedef struct {
     uart_rx_cb_t rx_cb;     /**< data received interrupt callback */
     void *arg;              /**< argument to both callback routines */
+    uint8_t data_mask;      /**< mask applied to the data register */
```

Comment:
Nooo, this is bad. Do not change the **global** interface to integrate some STM specific value!

Instead, simply define a custom type inside the `stm32_common/periph/uart.c`, e.g.
```c
typedef struct {
    uart_isr_ctx_t base;
    uint8_t data_mask;
} isrctx_t; // name does not matter, as it is only local to this file...

static isrctx_t _isr_ctx;
```
or maybe even better, define an inline struct in the uart.c file
```c
static struct {
    uart_rx_cb_t rx_cb;
    void *arg;
    uint8_t data_mask;
} _isr_ctx;
```

Reply:
You're right. I've created a local structure with `data_mask` field.